### ISSUE TITLE
No error generated on render failure due to bulky html

### ISSUE THREAD

**Issue by Codewolf** [dompdf/dompdf#1202]
After spending a good 4 hours trying to figure out why a blank page and no errors were reported after trying to generate a PDF from a large (bulky/nested tables) HTML structure i found that it was due to PHP exceeding its memory limit. Increasing this limit allowed me to render the PDF file correctly.

Issue: No error generated on failure to render bulky HTML string.
Expected: On failure an exception is thrown/error is logged to the error log.
Actual: White page is displayed, no further code execution happens as if die() has been called.

**Comment by bsweeney** [comment]
This is probably not an issue with dompdf specifically but a server configuration problem where memory exhaustion is happening outside dompdf (e.g. see [this question on SO](http://stackoverflow.com/q/37262152/264628)). I don't know that we can do anything about that, per se, unless you have evidence that it was specifically a dompdf problem.

I think it would be useful to add a low-memory-profile execution mode, but I'm not sure when this will happen. Not to mention there will likely be a potentially significant trade-off in rendering time for large document.

**Comment by Codewolf** [comment]
I Agree that the issue is not specifically a dompdf error, more that the error should be caught and an exception thrown as this allows the developer to watch out for this specific type of issue.

Another option is a throwing a "notice" or "warning" on trying to render a large document/lots of html as this may then discourage the end user.  Unfortunately while in my particular use case i would have to increase the memory size anyway due to the requirements of the html documents it may be worth adding in?

**Comment by bsweeney** [comment]
If you can track down the source of the issue and it's something we can catch then throwing an exception would definitely be useful. What I mean to point out in my previous comment is that I suspect that no catchable exception is thrown, e.g. PHP ran until the host process (such as apache) ran out of memory or otherwise crashed. In this situation there's not really an exception to catch.

However, if there's something in the crash that points to a catchable exception then definitely we should handle it.
